service/src/main/java/com/commercetools/util/HttpRequestUtil.java

```diff
@@ -79,10 +86,19 @@
                     .setConnectTimeout(REQUEST_TIMEOUT)
                     .build())
             .setRetryHandler(HTTP_REQUEST_RETRY_ON_SOCKET_TIMEOUT)
+            .setConnectionManager(buildDefaultConnectionManager())
```

Comment:
Maybe unrelated to this PR, but I don't understand this comment up there: 
`// empty implementation, we just need to use protected constructor`. Maybe make try to clarify/elaborate more?

Reply:
well, this is an inline class extension/implementation. We need to use absolutely same functionality as `DefaultHttpRequestRetryHandler`, but we can't call just `new DefaultHttpRequestRetryHandler(RETRY_TIMES, false, Arrays.asList(                    UnknownHostException.class, SSLException.class))` because this constructor is protected. 

Extending `DefaultHttpRequestRetryHandler` as a separate class also looks redundant because it is used only as an one instance and only once.

This i decided to extend/implement class inline, but we actually don't need to extend functionality (like add/override methods) thus the body `{}` could be left empty. But this could be confusing from first view. That's why i decided to put one short comment why the body is empty.

Honestly, i don't know how to describe this in a short comment....

Reply:
Thanks for the explanation. As discussed, please add short comment about anonymous extension.

Reply:
e76ec38e